blobfuse/fileapis.cpp

```diff
@@ -12,6 +12,53 @@ std::deque<file_to_delete> cleanup;
 std::mutex deque_lock;
 std::shared_ptr<gc_cache> g_gc_cache;
 
+int DownloadFileToDisk(std::string pathString, std::string mntPathString, BfsFileProperty blob_property, bool is_delayed)
+{
+    errno = 0;
+    time_t last_modified = {};
+
+    std::string disk_path = mntPathString;
+
+    syslog(LOG_DEBUG, "Starting download of file  %s", pathString.c_str());
+    long int size = storage_client->DownloadToFile(pathString.substr(1), disk_path, last_modified);
+    if (errno != 0)
+    {
+        int storage_errno = errno;
+        syslog(LOG_ERR, "Failed to download blob into cache.  Blob name: %s, file name = %s, storage errno = %d.\n", pathString.c_str()+1, mntPathString.c_str(),  errno);
+
+        remove(mntPathString.c_str());
+        return 0 - map_errno(storage_errno);
+    }
+    else
+    {
+        syslog(LOG_INFO, "Successfully downloaded blob %s into file cache as %s.\n", pathString.c_str()+1, mntPathString.c_str());
+        g_gc_cache->addCacheBytes(mntPathString, size);
+    }
+
+    // preserve the last modified time
+    struct utimbuf new_time;
+    new_time.modtime = last_modified;
+    new_time.actime = 0;
+    utime(mntPathString.c_str(), &new_time);
+
+    if (is_delayed) {
+        if (storage_client->isADLS()) {
+            // preserve the last modified time
+            struct utimbuf new_time;
```

Comment:
just like above the below 3 lines are not required.

Reply:
It can be taken care in above step so will remove this redundant reset of mod time.